templates/sfn/ingest_sfn_definition.json.tpl

```diff
@@ -190,34 +190,22 @@
           }
         }
       },
-      "Next": "Convert array to object",
-      "ResultPath": null
-    },
-    "Convert array to object": {
-      "Type": "Pass",
-      "Next": "Merge array",
-      "Parameters": {
-        "allFolders": {
-          "archiveHierarchyFolders.$": "$.archiveHierarchyFolders",
-          "contentFolders.$": "$.contentFolders"
-        },
-        "contentAssets.$": "$.contentAssets"
-      }
-    },
-    "Merge array": {
-      "Type": "Pass",
       "Next": "Map over each Folder Id",
-      "Parameters": {
-        "allFolders.$": "$.allFolders.[*]*",
-        "contentAssets.$": "$.contentAssets",
-        "executionId.$": "$$.Execution.Name",
-        "workflowContextName": "Ingest OPEX (Incremental)",
-        "stagingPrefix.$": "States.Format('opex/{}', $$.Execution.Name)",
-        "stagingBucket": "${ingest_staging_cache_bucket_name}"
-      }
+      "ResultPath": null
     },
     "Map over each Folder Id": {
       "Type": "Map",
+      "Label": "MapOverEachFolderId",
+      "ItemReader": {
+        "Resource": "arn:aws:states:::s3:getObject",
+        "ReaderConfig": {
+          "InputType": "JSON"
+        },
+        "Parameters": {
+          "Bucket.$": "$.folders.bucket",
+          "Key.$": "$.folders.key"
+        }
+      },
       "ItemsPath": "$.allFolders",
```

Comment:
We need to remove `ItemsPath` as we're now getting our data from S3 instead of the state data. The same will need doing on the asset map.